Marlin/Conditionals.h

```diff
@@ -628,7 +628,7 @@
   #define HAS_Z_MAX (PIN_EXISTS(Z_MAX))
   #define HAS_Z2_MIN (PIN_EXISTS(Z2_MIN))
   #define HAS_Z2_MAX (PIN_EXISTS(Z2_MAX))
-  #define HAS_Z_PROBE (PIN_EXISTS(Z_MIN_PROBE))
+  #define HAS_Z_MIN_PROBE_PIN (PIN_EXISTS(Z_MIN_PROBE))
```

Comment:
It reflects the naming of the pin. If the pin was called `PROBE_PIN` then by convention, this would be `HAS_PROBE` (but you can see why that's confusing, given the naming of some other options.) It's because of this confusion that I even added `_PIN` to the end in contravention.